tests/system_health/test_watchdog.py

```diff
@@ -16,14 +16,34 @@
 
 @pytest.fixture
 def pause_orchagent(duthost):
-    # find orchagent pid
-    pid = duthost.shell(
-                    r"pgrep orchagent",
-                    module_ignore_errors=True)['stdout']
-    logger.info('Get orchagent pid: {}'.format(pid))
-
-    # pause orchagent and clear syslog
-    duthost.shell(r"sudo kill -STOP {}".format(pid), module_ignore_errors=True)
+    pid = None
+    retry = 3
+    while True:
+        retry -= 1
+        # find orchagent pid
+        pid = duthost.shell(
```

Comment:
>pid [](http://example.com/codeflow?start=8&length=3)

Please post-check pid is valid. For example, if we lower orchagent privilege and it is not run by root.

Reply:
Fixed, change to pidof command and check command status.